core/src/net/network_thread.rs

```diff
@@ -610,7 +610,6 @@ impl NetworkThread {
             ChannelState::Closed(_, _) => {
                 trace!(self.log, "Connection shutdown gracefully");
                 self.notify_connection_state(channel.address(), ConnectionState::Closed);
-                self.drop_channel(channel);
```

Comment:
Why aren't we doing this here anymore? Did it move somewhere else?

Reply:
The ConnectionState::Closed is Ack'ed by the Dispatcher and then it's dropped, it should've never been there.